src/controllers/dynakube/connectioninfo/reconciler.go

```diff
@@ -33,7 +33,7 @@ func NewReconciler(ctx context.Context, clt client.Client, apiReader client.Read
 		dynakube:     dynakube,
 		scheme:       scheme,
 		dtc:          dtc,
-		timeProvider: timeprovider.New(),
+		timeProvider: timeprovider.New().SetToNow(),
```

Comment:
I don't think you need to "lock-in" the timestamp here. The time provider is used to check the current time against the timestamp of the last connection info request. An accurate "now" should be used.

Reply:
SetTo removed